app/extras/clients/base.rb

```diff
@@ -107,11 +107,12 @@ def post_content!(event)
   end
 
   def serialized_event(event)
-    begin
-      "#{self.class.name.demodulize}::#{event.kind.classify}Serializer".constantize
-    rescue NameError
-      "#{self.class.name.demodulize}::BaseSerializer".constantize
-    end.new(event, root: false).as_json
+    serializer = [
+      "#{self.class.name.demodulize}::#{event.kind.classify}Serializer",
```

Comment:
This is a imo slightly better approach to this, which cascades through specific to general class names. So, if we're doing a `poll_expired` event, we start with
```
Microsoft::PollExpiredSerializer
```
then, if we can't find that, we look for the eventable type
```
Microsoft::PollSerializer
```
and, failing those, the base serializer
```
Microsoft::BaseSerializer
```

This allows us to be super flexible in specifying the behaviour for each event, without a bunch of empty classes that don't do much.

Reply:
This is nice, but I think that I'd rather throw an exception if we don't have the correct serializer for the event.